### DISCUSSION TITLE
Trivy returns 2 different results on same package and same CVE

### DISCUSSION THREAD

**Discussion by tayalrishabh96** [aquasecurity/trivy#5964] [Bugs]
### Description

`FIXED_IN_VERSION` differs for same package and same CVE id for 2 different images say `i1` and `i2`.
In `i1` Trivy shows `FIXED_IN_VERSION` let's say `v2` where as in `i2` there is no `FIXED_IN_VERSION`.   
**NOTE :** i2 is patched version of i1.
PS : I'm patching the image using copacetic tool. [https://project-copacetic.github.io/copacetic/website/installation]

### Desired Behavior

My straight forward assumption is if there is a CVE in same package in 2 different images then `CURRENT_VERSION` and `FIXED_IN_VERSION` should not differ across both images.

### Actual Behavior

`FIXED_IN_VERSION` differs for same package and same CVE id for 2 different images.

### Reproduction Steps

```bash
1. Scan any image (i1) for instance : nginx:1.21.6 using trivy.
2. Patch the image (i1) using copa(https://project-copacetic.github.io/copacetic/website/quick-start), let's say patched image is (i2) / nginx:1.21.6-patched.
3. Scan the patched image (i2) again using trivy.
4. Compare both the results... 
Analysis : 
CVE-2023-44487 -> Package : libnghttp2-14
In 1st Trivy result a fixed in version was coming for above package but after patching the image from copa in 2nd Trivy result there was no fixed in version available for same CVE in same package.
```

### Target

Container Image

### Scanner

Vulnerability

### Output Format

None

### Mode

Standalone

### Debug Output

```bash
NA
```

### Operating System

macOS Monterey

### Version

```bash
0.46.1
```

### Checklist

- [X] Run `trivy image --reset`
- [X] Read [the troubleshooting](https://aquasecurity.github.io/trivy/latest/docs/references/troubleshooting/)

**Comment by DmitriyLewen** [comment]
Hello @tayalrishabh96 
Thanks for your report!

Can you rescan both images with the `-f json --list-all-pkgs` flags and send me the reports?

Regards, Dmitriy

**Reply by tayalrishabh96** [reply]
Hey @DmitriyLewen 
PFA: reports for both the images i) nginx:1.21.6 and ii) patched version of nginx:1.21.6
i) [nginx-1.21.6.json](https://github.com/aquasecurity/trivy/files/14082335/nginx-1.21.6.json)
ii) [nginx-patched.json](https://github.com/aquasecurity/trivy/files/14082337/nginx-patched.json)

**Reply by DmitriyLewen** [reply]
Thanks a lot!

`nginx-patched.json` file doesn't contain vulnerabilities. Do you see multiple results for `table` and `json` format?
